test/test-udp-send-ex.c

```diff
@@ -0,0 +1,258 @@
+                       const uv_buf_t* buf,
+                       const struct sockaddr* addr,
+                       unsigned flags) {
+  CHECK_HANDLE(handle);
+  ASSERT(flags == 0);
+  
+  fprintf(stderr, "client received\n");
+
+  if (nread < 0) {
+    ASSERT(0 && "unexpected error");
+  }
+
+  if (nread == 0) {
+    /* Returning unused buffer. Don't count towards cl_recv_cb_called */
+    ASSERT(addr == NULL);
+    return;
+  }
+
+  ASSERT(addr != NULL);
+  ASSERT(nread == 4);
+  ASSERT(!memcmp("PONG", buf->base, nread));
+
+  cl_recv_cb_called++;
+  uv_close((uv_handle_t*) handle, close_cb);
+}
+
+
+static void cl_send_cb(uv_udp_send_t* req, int status) {
+  int r;
+
+  ASSERT(req != NULL);
+  ASSERT(status == 0);
+  CHECK_HANDLE(req->handle);
+  
+  r = uv_udp_recv_start(req->handle, alloc_cb, cl_recv_cb);
+  ASSERT(r == 0);
+
+  cl_send_cb_called++;
+}
+
+
+static void sv_send_cb(uv_udp_send_t* req, int status) {
+  ASSERT(req != NULL);
+  ASSERT(status == 0);
+  CHECK_HANDLE(req->handle);
+  
+  uv_close((uv_handle_t*) req->handle, close_cb);
+  free(req);
+
+  sv_send_cb_called++;
+}
+
+
+static void sv_recv_cb(uv_udp_t* handle,
+                       ssize_t nread,
+                       const uv_buf_t* rcvbuf,
+                       const struct sockaddr* addr,
+                       unsigned flags) {
+  uv_udp_send_t* req;
+  uv_buf_t sndbuf;
+  int r;
+  
+  fprintf(stderr, "test\n");
+
+  if (nread < 0) {
+    ASSERT(0 && "unexpected error");
+  }
+
+  if (nread == 0) {
+    /* Returning unused buffer. Don't count towards sv_recv_cb_called */
+    ASSERT(addr == NULL);
+    return;
+  }
+
+  CHECK_HANDLE(handle);
+  ASSERT(flags == 0);
+
+  ASSERT(addr != NULL);
+  ASSERT(nread == 4);
+  ASSERT(!memcmp("PING", rcvbuf->base, nread));
+
+  /* FIXME? `uv_udp_recv_stop` does what it says: recv_cb is not called
+    * anymore. That's problematic because the read buffer won't be returned
+    * either... Not sure I like that but it's consistent with `uv_read_stop`.
+    */
+  if (sv_recv_cb_called > 1) {
+    r = uv_udp_recv_stop(handle);
+    ASSERT(r == 0);
+  }
+
+  req = malloc(sizeof *req);
+  ASSERT(req != NULL);
+  
+  fprintf(stderr, "request came from path %s\n", ((struct sockaddr_un*) addr)->sun_path);
+
+  sndbuf = uv_buf_init("PONG", 4);
+  r = uv_udp_send_ex(req, handle, &sndbuf, 1, addr, sizeof(struct sockaddr_un), sv_send_cb);
+  ASSERT(r == 0);
+
+  sv_recv_cb_called++;
+}
+
+static const char* server_socket_path = "./test_socket_server";
+static const char* client_socket_path = "./test_socket_client";
+
+TEST_IMPL(udp_send_ex) {
+  struct sockaddr_un server_addr;
+  struct sockaddr_un client_addr;
+  uv_udp_send_t req;
+  uv_buf_t buf;
+  int r;
+  int server_sock;
+  int client_sock;
+  
+  unlink(client_socket_path);
+  unlink(server_socket_path);
+  
+  server_sock = socket(AF_UNIX, SOCK_DGRAM, 0);
+  ASSERT(server_sock >= 0);
+  
+  memset(&server_addr, 0, sizeof(struct sockaddr_un));
```

Comment:
```suggestion
  memset(&server_addr, 0, sizeof(server_addr));
```
Ditto on the lines below. I won't point out individual instances.